tests/test_user_flows.py

```diff
@@ -1,36 +1,202 @@
+        ('ad-saml', 'yes'),
+        ('gds-google', 'no'),
+        ('co-digital', 'no'),
+        ('ad-saml', 'no'),
+    ])
+    def test_confirm_idp(self, app_, idp_id, choice):
+        url = url_for('main.confirm_idp')
+        data = {
+            'confirm': choice
+        }
+
+        with app_.test_client() as c:
+            with c.session_transaction() as sess:
+                sess['suggested_idp'] = idp_id
+
+            assert c.get(url).status_code == 200
+
+            resp = c.post(url, data=data)
+            assert resp.status_code == 302
+
+            if choice == 'yes':
+                assert resp.location.endswith(
+                    url_for('broker.auth', idp_hint=idp_id))
+
+            if choice == 'no':
+                assert resp.location.endswith(
+                    url_for('main.request_email_address'))
+
+    @pytest.mark.parametrize("idp_id", [
+        'gds-google',
+        'co-digital',
+        'ad-saml',
+    ])
+    def test_select_dept(self, app_, idp_id):
+        url = url_for('main.select_dept')
+        data = {
+            'dept': idp_id
+        }
+
+        with app_.test_client() as c:
+            assert c.get(url).status_code == 200
+
+            resp = c.post(url, data=data)
+            assert resp.status_code == 302
+            assert resp.location.endswith(url_for('main.to_idp'))
+
+    @pytest.mark.parametrize("email, idp, name", email_idp_name)
+    def test_confirm_dept(self, app_, email, idp, name):
+        url = url_for('main.confirm_dept')
+        data = {
+            'confirm': 'yes'
+        }
+
+        with app_.test_client() as c:
+            with c.session_transaction() as sess:
+                sess['suggested_idp'] = idp
+                sess["department_name"] = name
+                sess["email_address"] = email
+
+            resp = c.get(url)
+            assert resp.status_code == 200
+            content = str(resp.data)
+            assert email in content
+            assert name in content
 
-    @pytest.mark.xfail
-    def test_change_email_address(self):
-        assert False
+            resp = c.post(url, data=data)
+            assert resp.status_code == 302
+            assert resp.location.endswith(url_for('main.to_idp'))
 
-    @pytest.mark.xfail
-    def test_select_idp(self):
-        assert False
+    @pytest.mark.parametrize("idp_id", [
+        'gds-google',
+        'co-digital',
+        'ad-saml',
+    ])
+    def test_interstitial_to_idp(self, app_, idp_id):
+        url = url_for('main.to_idp')
 
-    @pytest.mark.xfail
-    def test_confirm_idp(self):
-        assert False
+        with app_.test_client() as c:
+            with c.session_transaction() as sess:
+                sess['suggested_idp'] = idp_id
 
-    @pytest.mark.xfail
-    def test_select_dept(self):
-        assert False
+            resp = c.get(url)
+            assert resp.status_code == 200
+            self.assertMetaRefresh(resp, 5, url_for('broker.auth',
```

Comment:
This 5 is a magic number. This should be a named constant in the test, or maybe from configuration, so that the test and the app can share it and it won't break tests later if the value is changed.